pysph/tools/ipy_viewer.py

```diff
@@ -540,3 +541,294 @@ def _save_figure_handler(self, change):
                 )
                 break
         self._widgets.save_figure.value = ""
+
+
+class ParticleArrayWidgets3D(object):
+
+    def __init__(self, particlearray, particles_type):
```

Comment:
what is `particles_type`?  If you mean or the `name` of the array?

Reply:
It's the same as we were using for 2d. It takes the values - _fluid_, _boundary_, _inlet_, etc.

Reply:
Then I would prefer that you change both the variable names to something more meaningful like `array_name` or even just `name`.  That said, you don't need to pass the attribute at all as the name is always available from the particle array itself.  `particlearray.name` will give you the same thing.  Can you please fix this?  Thanks.